### PR TITLE
Update AnswerFormatHelp to use the new helpLink method and deprecate the macro

### PR DESCRIPTION
**Pull request by drgrice1**
The helpLink macro is better and has more functionality.  This macro is not needed anymore.  It is updated to just call helpLink.  This means that a knowl will be added instead of a link to a page that opens in a new window with lots of invalid html.

Note that a new help file for matrices was added that was in the AnswerFormatHelp.pl macro, but did not have an equivalent in the helpLink method.  Also the default help text is added to the helpLink method so things look the same for those using the now deprecated macro.

I honestly think that instead of mixing all of the OPL macros in with the existing PG macros, almost all of them should have been moved into deprecated.

### CODE REVIEW

**Comment by drgrice1** [comment]
By the way, when I say that `helpLink` is new, I mean new as in 19 years ago.  The method was added in 2004.  Of course it has been updated since then.  AnswerFormatHelp in the mean time has had some updates, but still outputs invalid html and uses bad practice both in the javascript implementation and in what it does by opening a new window in the way it does.

**Comment by pstaabp** [comment]
This might be a good time to add some POD to the helpLink sub as well.

File: macros/core/PGbasicmacros.pl [hunk 1]
```diff
@@ -2656,39 +2656,43 @@ sub iframe {
...
+		'angle'     => [ 'Entering-Angles.html',       'help (angles)' ],
+		'decimal'   => [ 'Entering-Decimals.html',     'help (decimals)' ],
+		'equation'  => [ 'Entering-Equations.html',    'help (equations)' ],
+		'exponent'  => [ 'Entering-Exponents.html',    'help (exponents)' ],
+		'formula'   => [ 'Entering-Formulas.html',     'help (formulas)' ],
+		'fraction'  => [ 'Entering-Fractions.html',    'help (fractions)' ],
+		'inequalit' => [ 'Entering-Inequalities.html', 'help (inequalities)' ],
+		'limit'     => [ 'Entering-Limits.html',       'help (limits)' ],
+		'log'       => [ 'Entering-Logarithms.html',   'help (logarithms)' ],
+		'matri'     => [ 'Entering-Matrices.html',     'help (matrices)' ],
```
[the remark this answers is not in the document]
Comment (pstaabp) [reply macros/core/PGbasicmacros.pl:2675]: Never mind.  This will match both matrix and matrices.

[the remark this answers is not in the document]
Comment (drgrice1) [reply macros/core/PGbasicmacros.pl:2675]: Yeah, that was not a typo.  It took me a bit to figure out why it was that way also.  It is is the same for 'inequalit'.

**Comment by drgrice1** [comment]
@pstaabp:  I added POD for the helpLink method.

**Review by pstaabp — APPROVED** [review]

**Pull request merged**